clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java

```diff
@@ -109,6 +114,25 @@ private Replacer(
     return replacedArtifacts.build();
   }
 
+  private Predicate createReplaceFilterPredicate(List<String> replacePath, String name) {
+    return ctx -> {
+      Map<String, Object> ctxMap = ctx.item(Map.class);
```

Comment:
I think here we can have JSONPath do this lookup for us so we don't need to handle looking up the various keys at each level.

Something like this should directly get you the value at the input replace path (where the `@.` that's prepended is JSONPath for "relative to the current node"):
```
String path = String.join(".", replacePath);
ValueNode node = ValueNode.toValueNode("@." + path).asPathNode().evaluate(ctx);

// Then you can check if it's a String with
node.isStringNode()
// And get the value with
String value = node.asStringNode().getString()
```
(Given we're re-joining the `List<String> replacePath` it might make sense if you go with the above to leave it as it was initially, as a dot-separated JsonPath query.)

The main reason I suggest this is just that given we're invested in JsonPath in this class I think we should try to encapsulate as much as possible behind it; by doing this we can avoid some manual iteration and unchecked casts that the library handles.

But overall I like this solution a lot better than using a regex as re-using `nameFromReference` will keep behavior consistent with any future changes. 💯

Reply:
Thanks for the response. Your suggestion is much cleaner, I've went ahead and updated the PR.